### PR TITLE
Add unit test coverage for events expansion

### LINKED ISSUE #105276: Unit tests coverage for events expansion
Follow up from https://github.com/kubernetes/kubernetes/pull/104507

We should cover events expansion - both for
- core v1: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/client-go/kubernetes/typed/core/v1
- events v1beta: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/client-go/kubernetes/typed/events/v1beta1
with unit tests

@wzshiming - can you please take care of it?

### PR DESCRIPTION
**Pull request by wzshiming**
#### What type of PR is this?

/kind cleanup
/kind feature

#### Which issue(s) this PR fixes:

Fixes #105276

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

/cc @wojtek-t

### CODE REVIEW

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	"testing"
+
+	v1 "k8s.io/api/events/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/rest/fake"
+)
+
+func Test_events_CreateWithEventNamespace(t *testing.T) {
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go:31]: Let's avoid underscore.

Let's just name it TestCreateWithEventNamespace.

[Same everywhere below]

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+		Client: fake.CreateHTTPClient(func(request *http.Request) (*http.Response, error) {
+			resp := &http.Response{
+				StatusCode: http.StatusOK,
+				Body:       request.Body,
+			}
+			return resp, nil
+		}),
+		NegotiatedSerializer: scheme.Codecs.WithoutConversion(),
+	}
+	type fields struct {
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go:48]: Let's remove this type and inline it in the test.

In fact - You're always using the same client - so let's just not even put it into test right now.

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+		client rest.Interface
+		ns     string
+	}
+	type args struct {
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go:52]: Let's remove this type - just directly inline event below.

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "default",
+		},
+	}
+	cli := &fake.RESTClient{
+		Client: fake.CreateHTTPClient(func(request *http.Request) (*http.Response, error) {
+			resp := &http.Response{
+				StatusCode: http.StatusOK,
+				Body:       request.Body,
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go:42]: Seems misleading - let's remove. Let's rely on the StatusCode only.

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+			},
+			args: args{
+				event: &data,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := &events{
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go:87]: Can you use newEvents() instead?

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("CreateWithEventNamespace() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_events_PatchWithEventNamespace(t *testing.T) {
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go:103]: Same comments as above for this test and all below too.

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	restfake "k8s.io/client-go/rest/fake"
+)
+
+func TestCreateWithEventNamespace(t *testing.T) {
+	data := v1.Event{
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go:34]: Let's call it event maybe. And let's make it pointer already:

event := &v1.Event{ ...

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+				Body:       io.NopCloser(strings.NewReader("{}")),
+			}
+			return resp, nil
+		}),
+		NegotiatedSerializer: scheme.Codecs.WithoutConversion(),
+	}
+
+	tests := []struct {
+		name    string
+		client  rest.Interface
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go:53]: Let's remove this from the test definition  - you're using the same thing everywhere.

File: staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("CreateWithEventNamespace() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestPatchWithEventNamespace(t *testing.T) {
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/kubernetes/typed/core/v1/event_expansion_test.go:89]: Same in other tests.

**Review by wojtek-t — COMMENTED** [review]
Please also squash commits.

**Comment by wzshiming** [comment]
> Please also squash commits.

These two commits are irrelevant, one is to add tests, the other is to update interface comments, is it clearer to separate

**Comment by wojtek-t** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**